plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsReader.java

```diff
@@ -325,8 +327,16 @@ protected Long computeNext(Long previous)
             {
                 requireNonNull(previous, "previous is null");
                 @Nullable
-                Long parentId = icebergTable.snapshot(previous).parentId();
-                return parentId;
+                Snapshot snapshot = icebergTable.snapshot(previous);
+                if (snapshot == null) {
+                    // Snapshot referenced by `previous` is expired from table history
+                    return null;
+                }
+                if (snapshot.parentId() == null) {
+                    // Snapshot referenced by `previous` had no parent.
+                    return null;
+                }
+                return verifyNotNull(snapshot.parentId(), "snapshot.parentId()");
```

Comment:
`verifyNotNull` seem redundant here - given that we check above
```
if (snapshot.parentId() == null)
```

Reply:
yes, obviously.

it's here for readability & future proofing that we do not return end-of-data on this line accidentally. 

otherwise i could just have
```
return snapshot.parentId();
```

without preceding `if` on that value